fast_matched_filter/fast_matched_filter.py

```diff
@@ -92,11 +92,30 @@ def matched_filter(templates, moveouts, weights, data, step, arch='cpu'):
         print("Compiled library for {} not loaded; exiting!".format(arch))
         return
 
-    n_templates = np.int32(templates.shape[0])
-    n_stations = np.int32(data.shape[0])
-    n_components = np.int32(data.shape[1])
+    if templates.ndim == 2:
+        n_templates = np.int32(1)
+    elif templates.ndim == 3 and data.ndim == 3:
+        n_templates = np.int32(1)
+    else:
+        n_templates = np.int32(templates.shape[0])
     n_samples_template = np.int32(templates.shape[-1])
+    
+    n_stations = np.int32(data.shape[0])
     n_samples_data = np.int32(data.shape[-1])
+    if data.ndim == 2:
+        n_components = np.int32(1)
+        data = data.reshape(n_stations, n_components, n_samples_data)
+    else:
+        n_components = np.int32(data.shape[1])
+
+    # reshape inputs if necessary
+    if templates.shape != (n_templates, n_stations, n_components, n_samples_template):
```

Comment:
Can this be:
```python
if templates.shape == (n_templates, n_stations * n_components, n_samples_template):
    templates.reshape(n_templates, n_stations,
                                   n_components, n_samples_template)
assert templates.shape == (n_templates, n_stations, n_components, n_samples_template)
```
then you are only reshaping for the case that you expect rather than potentially reshaping some super weird shaped data that the user has provided. I often provide the wrong shape data to things and really appreciate being told when I'm doing something unexpected rather than having to sift through source code to workout why I'm not getting the result I expect!

Reply:
This does seem like a more bulletproof way to do this. I'll shape all of the current if statements to this format? (see the pun?!)